grakn/service/Session/util/Communicator.py

```diff
@@ -0,0 +1,112 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import six
+from collections import deque
+from six.moves import queue
+from grakn.exception.GraknError import GraknError
+
+
+class IterationResolver(six.Iterator):
+    def __init__(self, communicator, request, is_last_response=lambda x: True):
```

Comment:
can we not have a default for the last arg so it's always required here?

Reply:
Hmm, I had it like that before but this felt a bit more concise, will revert